docs/madr/decisions/057-rework-mesh-routes-matching.md

```diff
@@ -0,0 +1,382 @@
+- [Policy matching with real resources](https://github.com/kumahq/kuma/pull/10726/files)
+
+### API
+
+New API for policy that selects Mesh*Route
+
+```yaml
+type: MeshTimeout
+name: timeouts-on-route
+mesh: default
+spec:
+  targetRef:
+    kind: Mesh | MeshSubset # either Mesh, MeshSubset or MeshGateway
+  to:
+    - targetRef:
+        kind: MeshHTTPRoute | MeshTCPRoute # either MeshHTTPRoute or MeshTCPRoute
+        name: route-to-backend
+      default:
+        http:
+          requestTimeout: 5s
+          streamIdleTimeout: 1h
+```
+
+specifying `Mesh` in topLevel targetRef will be optional (this probably will be most used case). Minimal policy, targeting
+`MeshHTTPRoute` will look like this:
+
+```yaml
+type: MeshTimeout
+name: timeouts-on-route
+mesh: default
+spec:
+  to:
+    - targetRef:
+        kind: MeshHTTPRoute
+        name: route-to-backend
+      default:
+        http:
+          requestTimeout: 5s
+          streamIdleTimeout: 1h
+```
+
+To introduce this change we need to deprecate specifying Mesh*Route in topLevel targetRef.
+
+Should we allow selecting MeshGateway in topLevel targetRef? probably this is obsolete as Route will select gateway, and we
+want to apply policy on a route. Is there a need to make it more specific than route? 
+
+### Merging and applying configurations
+
+Let's start with an example:
+
+```yaml
+# Producer route in backend-ns namespace
+apiVersion: kuma.io/v1alpha1
+kind: MeshHTTPRoute
+metadata:
+  name: route-to-backend
+  namespace: backend-ns
+  labels:
+    kuma.io/mesh: default
+spec:
+  to:
+  - targetRef:
+      kind: MeshService
+      name: backend
+    rules:
+    - matches:
+      - path:
+          type: PathPrefix
+          value: "/slow-endpoint"
+      default:
+        backendRefs:
+          - kind: MeshService
+            name: backend
+---
+# Producer MeshTimeout targeting whole MeshService
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: timeout-on-backend-service
+  namespace: backend-ns
+  labels:
+    kuma.io/mesh: default
+spec:
+  to:
+  - targetRef:
+      kind: MeshService
+      name: backend
+    default:
+      http:
+        requestTimeout: 2s
+        streamIdleTimeout: 1h
+---
+# MeshTimeout targeting producer route
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: timeout-on-backend-route
+  namespace: backend-ns
+  labels:
+    kuma.io/mesh: default
+spec:
+  to:
+  - targetRef:
+      kind: MeshHTTPRoute
+      name: route-to-backend
+    default:
+      http:
+        requestTimeout: 10s
+```
+
+As described in targeting real resources MADR, while merging policies we will create `ResourceRules map[UniqueResourceKey]ResourceRule` 
+field that holds configuration for a given resource identified by `UniqueResourceKey`. So when user creates policy that has 
+real resource in `to` section we will create confs by resources
```

Comment:
This explanation feels kind of incomplete or has missing context, what Dataplane does this apply to? It might be a lot clearer in fact to just show the resulting config for a given Dataplane.

Reply:
I've added more context to this section